druid/src/widget/svg.rs

```diff
@@ -0,0 +1,284 @@
+                        usvg::PathSegment::MoveTo { x, y } => {
+                            let x = (x * scale) + offset.x;
+                            let y = (y * scale) + offset.y;
+                            path.move_to((x, y));
+                        }
+                        usvg::PathSegment::LineTo { x, y } => {
+                            let x = (x * scale) + offset.x;
+                            let y = (y * scale) + offset.y;
+                            path.line_to((x, y));
+                        }
+                        usvg::PathSegment::CurveTo {
+                            x1,
+                            y1,
+                            x2,
+                            y2,
+                            x,
+                            y,
+                        } => {
+                            let x1 = (x1 * scale) + offset.x;
+                            let y1 = (y1 * scale) + offset.y;
+                            let x2 = (x2 * scale) + offset.x;
+                            let y2 = (y2 * scale) + offset.y;
+                            let x = (x * scale) + offset.x;
+                            let y = (y * scale) + offset.y;
+
+                            path.curve_to((x1, y1), (x2, y2), (x, y));
+                        }
+                        usvg::PathSegment::ClosePath => {
+                            path.close_path();
+                        }
+                    }
+                }
+                match &p.fill {
+                    Some(fill) => {
+                        let brush = color_from_fill(fill);
+                        paint_ctx.fill(path.clone(), &brush);
+                    }
+                    None => {}
+                }
+
+                match &p.stroke {
+                    Some(stroke) => {
+                        let brush = color_from_stroke(stroke);
+                        paint_ctx.stroke(path.clone(), &brush, stroke.width.value());
+                    }
+                    None => {}
+                }
+            }
+            usvg::NodeKind::Defs => {
+                // TODO: implement defs
+            }
+            _ => {
+                // TODO: handle more of the SVG spec.
+                error!("{:?} is unimplemented", n.clone());
+            }
+        }
+    }
+}
+
+#[cfg(feature = "svg")]
+fn color_from_stroke(stroke: &usvg::Stroke) -> Color {
+    match stroke.paint {
+        usvg::Paint::Color(c) => {
+            Color::rgb8(c.red, c.green, c.blue).with_alpha(stroke.opacity.value())
+        }
+        _ => {
+            //TODO: implement link
+            error!("We don't support Paint::Link yet, so here's some pink.");
+            Color::rgb8(255, 192, 203)
+        }
+    }
+}
+
+#[cfg(feature = "svg")]
+fn color_from_fill(fill: &usvg::Fill) -> Color {
+    match fill.paint {
+        usvg::Paint::Color(c) => {
+            Color::rgb8(c.red, c.green, c.blue).with_alpha(fill.opacity.value())
```

Comment:
these two impls share the bulk of their code, might be room for a helper function?

I also might consider making these impls of the `From` trait, but I totally don't mind doing it like this for one-offs.